relays/ethereum/src/rpc.rs

```diff
@@ -0,0 +1,338 @@
+// Copyright 2019-2020 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! RPC Module
+
+#![allow(dead_code)]
+#![allow(unused_variables)]
+#[warn(missing_docs)]
+use std::result;
+
+use crate::ethereum_client::{CallRequest, EthereumConnectionParams};
+use crate::ethereum_types::{
+	Address as EthAddress, Bytes, EthereumHeaderId, Header as EthereumHeader, Receipt, SignedRawTx,
+	TransactionHash as EthereumTxHash, H256, U256, U64,
+};
+use crate::rpc_errors::{EthereumNodeError, RpcError};
+use crate::substrate_client::SubstrateConnectionParams;
+use crate::substrate_types::{Hash as SubstrateHash, Header as SubstrateHeader, Number as SubBlockNumber};
+use crate::sync_types::HeaderId;
+
+use async_trait::async_trait;
+use bridge_node_runtime::UncheckedExtrinsic;
+use codec::{Decode, Encode};
+use jsonrpsee::common::Params;
+use jsonrpsee::raw::client::RawClient;
+use jsonrpsee::transport::http::HttpTransportClient;
+use serde_json;
+use sp_bridge_eth_poa::{EthereumHeadersApiCalls, Header as SubstrateEthereumHeader, RuntimeApiCalls};
+
+/// Proof of hash serialization success.
+const HASH_SERIALIZATION_PROOF: &'static str = "hash serialization never fails; qed";
+/// Proof of integer serialization success.
+const INT_SERIALIZATION_PROOF: &'static str = "integer serialization never fails; qed";
+/// Proof of bool serialization success.
+const BOOL_SERIALIZATION_PROOF: &'static str = "bool serialization never fails; qed";
+
+type Result<T> = result::Result<T, RpcError>;
+type GrandpaAuthorityList = Vec<u8>;
+
+jsonrpsee::rpc_api! {
+	Ethereum {
+		#[rpc(method = "eth_estimateGas")]
+		fn estimate_gas(call_request: Params) -> U256;
```

Comment:
Isn't it possible to have strictly typed arguments instead of `Params`?

Reply:
Yeah you're right, I'll update this